### ISSUE TITLE
[FEEDBACK] I'm not sure if I understand Instant playlists

### ISSUE THREAD

**Issue by elzorrorebelde** [NeptuneHub/AudioMuse-AI#472]
**Thank you for taking the time to provide feedback! This is incredibly valuable for fine-tuning the software.**

**1. What were you trying to do?**
I was looking to generate a playlist for weightlifting, your typical "aggresive" music playlist

**2. What did you like?**
I was able to use xAI's grok-4.3 model with AudioMuse

**4. Task Parameters You Used (Very Important!)**
If your feedback is related to an Analysis or Clustering task, please share the key parameters you used. This helps reproduce the context of your experience.

*   **Analysis Task:**
    *   `NUM_RECENT_ALBUMS`: 0
    *   Any other configuration used for clustering.
*   **Clustering Task:**
    *   `ENABLE_CLUSTERING_EMBEDDINGS`: True
    *   `CLUSTER_ALGORITHM`: kmeans
    *   `NUM_CLUSTERS_MIN / MAX`: 40 / 100
    *   `CLUSTERING_RUNS`: 1000
    *   `SCORE_WEIGHT_DIVERSITY` / `SCORE_WEIGHT_PURITY`: 2 / 1
    *   `AI_MODEL_PROVIDER`: OpenAI / OperRouter (xAI https://api.x.ai/v1/chat/completions)
    *   I kept the defaults, including `5` moods.

**5. Any Other Thoughts or Ideas?**

<details>

<summary>So, this is the request summary:</summary>

```
🎵 NEW MCP-BASED PLAYLIST GENERATION

Request: 'Aggressive music for heavy weightlifting, beast mode'

AI Provider: OPENAI

🤖 Using MCP Agentic Workflow for playlist generation

Target: 100 songs

Available tools: song_similarity, text_search, artist_similarity, song_alchemy, ai_brainstorm, search_database

Library: 4873 songs, 68 artists

============================================================

ITERATION 1/5

Current progress: 0/100 songs (collected 0)

============================================================

--- AI Decision (Iteration 1) ---

Using timeout: 300 seconds for OpenAI/Mistral request

OpenAI called 1 tools

--- Executing 1 Tool(s) ---

🔧 Tool 1/1: text_search

   Arguments: {

      "description": "Aggressive music for heavy weightlifting, beast mode",

      "get_songs": 200

}

   ✅ Retrieved 99 songs from database

   CLAP text search: 'Aggressive music for heavy weightlifting, beast mode'

   CLAP returned 99 songs

   Retrieved 99 songs from CLAP

   📊 Added 99 NEW unique songs

   🎵 Sample songs: 5/99

      1. Run, Pig, Run - Queens of the Stone Age

      2. Behemoth - Static‐X

      3. Spitfire - The Prodigy

      4. In the Hall of the Mountain King - The Offspring

      5. The Day Is My Enemy - The Prodigy

📈 Iteration 1 Summary:

   Songs added this iteration: 99

   Total songs now: 99/100 usable (collected 99)

============================================================

ITERATION 2/5

Current progress: 99/100 songs (collected 99)

============================================================

--- AI Decision (Iteration 2) ---

Using timeout: 300 seconds for OpenAI/Mistral request

OpenAI called 1 tools

--- Executing 1 Tool(s) ---

🔧 Tool 1/1: search_database

   Arguments: {

      "moods": [

            "aggressive"

      ],

      "get_songs": 200

}

   ✅ Retrieved 200 songs from database

   Found 200 songs matching moods: aggressive

   📊 Added 194 NEW unique songs

   🎵 Sample songs: 5/194

      1. Fíjate bien - Juanes

      2. My Own Way - Duran Duran

      3. Policeman - Chicago

      4. Skit #3 - Ye

      5. B‐Boy Bouillabaisse - Beastie Boys

📈 Iteration 2 Summary:

   Songs added this iteration: 194

   Total songs now: 209/100 usable (collected 293)

============================================================

ITERATION 3/5

Current progress: 209/100 songs (collected 293)

============================================================

✅ Target reached (209 usable songs)! Stopping.

🎨 Artist diversity: removed 84 excess songs from pool (max 5/artist)

📊 Pool: 293 collected → 209 after diversity cap → 100 in final playlist

🎵 Playlist ordered for smooth transitions (tempo/energy/key)

✅ SUCCESS! Generated playlist with 100 songs

   Total songs collected: 293

   Iterations used: 3/5

   Tools called: 2

📊 Tool Contribution (Collected → Final Playlist):

   • text_search(no filters): 99 collected → 48 in final playlist

   • search_database(moods=['aggressive']): 194 collected → 52 in final playlist
```
</details>

<details>

<summary>The songs? well...</summary>
```
Wishlist by Pearl Jam
Cacophony by blink‐182
I Love the Life I Live (I Live the Life I Love) by Muddy Waters
I Wanna Be Where You Are by Michael Jackson
Reflection Eternal by Nujabes
I Ain't the One by Lynyrd Skynyrd
Wish You Could Be Me by Korn
Niagara Falls by Chicago
Policeman by Chicago
Wearing the Inside Out by Pink Floyd
Autumn ’68 by Pink Floyd
My Own Way by Duran Duran
B‐Boy Bouillabaisse by Beastie Boys
Skit #3 by Ye
The Thin Ice by Pink Floyd
Intro by Billy Idol
Prelude to a Come Up by Cypress Hill
Dapper-Dan by Michael Jackson
Icing Sugar by The Cure
Conspiracy by Paramore
I Can Never Say Goodbye (instrumental) by The Cure
Rok‐Weiler by The Prodigy
The Bronze by Queens of the Stone Age
Hispanic Impressions by Queens of the Stone Age
Wonderful Woman by Chuck Berry
Reptile by Static‐X
Forty Ways by Static‐X
Behemoth by Static‐X
Spitfire by The Prodigy
La prisión by Maná
Just Good Friends by Michael Jackson
Thank You by Nujabes
Alright by Electric Light Orchestra
Song for the Dead by Queens of the Stone Age
Victimized by Linkin Park
Just a Girl by Pearl Jam
Are You Ready to Live? by Korn
Patri(di)ots by Marcin Przybyłowicz
Gone Away by The Offspring
Change the Skyline by Duran Duran
Swordfish Hotkiss Night (Eron Mezza remix) by Empire of the Sun
Swordfish Hotkiss Night by Empire of the Sun
Nuestro juramento by Juanes
Slave to the Rhythm by Michael Jackson
Sky Is Tumbling by Nujabes
The Heist by P.T. Adamczyk
Cigaro by System of a Down
Run, Pig, Run by Queens of the Stone Age
Get on Top by Red Hot Chili Peppers
Juiced Up by P.T. Adamczyk
Looking Out For #1 by The Offspring
MADEINHELL, PT. 2 by Kordhell
Dat Phonk by Kordhell
Embryo by Black Sabbath
…And Justice for All by Metallica
Need Some1 by The Prodigy
Three MC’s and One DJ by Beastie Boys
Can’t Get You More Pregnant by blink‐182
Narcissistic Cannibal by Korn
Dirty Window by Metallica
Cyberwildlife Park by Marcin Przybyłowicz
Give Me It (studio demo) by The Cure
Happy Like You by Empire of the Sun
Again & Again by Oliver Tree
Black Moon by Black Sabbath
No More Sorrow by Linkin Park
Neocon by The Offspring
Blessing It (remix) by Nujabes
Drone:Nodrone (instrumental) by The Cure
Burn the Obedient by Korn
Cyberninja by Marcin Przybyłowicz
Black Star by Static‐X
Adam Smasher by Paul Leonard‐Morgan
Brohemian Rhapsody by blink‐182
We Are the People (Shapeshifters vocal mix) by Empire of the Sun
In the Hall of the Mountain King by The Offspring
Ode (live at Freeman Coliseum, 1999) by Creed
Fíjate bien by Juanes
Мусорщики by P.T. Adamczyk
Hallelujah by Paramore
We Live Forever by The Prodigy
I Can’t Be With You (live in Toronto) by The Cranberries
Liberian Girl by Michael Jackson
Would? by Alice in Chains
Do It for Love by Daryl Hall & John Oates
The Day Is My Enemy by The Prodigy
Sick, Sick, Sick by Queens of the Stone Age
Good Morning by Ye
Soil by System of a Down
When They Come for Me by Linkin Park
Purify by Metallica
All Within My Hands by Metallica
Generational Divide by blink‐182
Xmas in the Booth by Ice Cube
Bubbles by System of a Down
Fuels the Comedy by Korn
A Very Special Leimert Park Xmas by Cookin’ Soul
In 3’s by Beastie Boys
A New Machine, Part 2 (2019 remix) by Pink Floyd
A Million Ways to Murder by Kordhell
```
</details>

Muddy Waters, Chicago, and Michael Jackson aren't known for making aggressive music.

Something like this happened too when making a playlist for studying or reading (music lacking of vocals, mostly instrumental); some songs did have vocals in them.

**Comment by rendyhd** [comment]
Hi @shackra !
I see two things.
1 - @NeptuneHub we don't have a cutoff on the mood filter in the mcp, and since it's a scale it'll match to broad
2 - @shackra aggressive in this context (besides the bug in 1) is a classifier, which leans more towards the metal side - like distorted guitars, harshness. It doesn't mean high tempo / energy what you sometimes expect in a gym session. So there is a bug, but I'm not sure that without the bug you get what you expect. I'm working on a Smart Search that should help you finetune use cases like this in #417

**Comment by NeptuneHub** [comment]
The AI can choose from several tools that already exist in AudioMuse-AI, plus database search and AI knowledge (which have sense with larger models). Its goal is to understand a query and combine multiple tools to produce results that a single tool couldn’t achieve alone.

The advantage is the ability to aggregate results from different tools. The downside is unpredictability due to AI behavior.

For specific requests like “aggressive rock songs,” it’s better to use dedicated tools (e.g., similar songs + filters like “aggressive” and genre) for more precise results.

Instant Playlist performs best with aggregated queries. For example, “songs like Iron Maiden, Metallica, and Deep Purple” can combine similarities across multiple artists.

It’s not perfect and I would like to dedicate time with @rendyhd to improve it, but the key point remains: it’s not designed for precise, single requests, it works better for broader, aggregated ones.

You can start experimenting using the example of query directly integrated in the functionality.

**Comment by NeptuneHub** [comment]
Hi,
in the image:
- ghcr.io/neptunehub/audiomuse-ai:pr-484

There is some improvement related to this made from @rendyhd 

Please when you have time give a try and let us know. Thanks!

**Comment by rendyhd** [comment]
confirmed to be fixed in #484

**Comment by NeptuneHub** [comment]
@shackra fix released in` v1.1.1` that is actually building, will be ready in around 1h.

Rendy discovered a wrong query on the database related to the featrure selection (like Happy, Sad, Aggresive, etc). With this at least the specific error when you search Aggressive Rock should be fixed.

Remain the fact that AI have his limit here and could allucinate especially on small model, but we will do our best to keep improving it.

**Comment by elzorrorebelde** [comment]
cool, I will fetch the latest commit and let you know (I don't use Docker for deploying this software on my machine)

**Comment by elzorrorebelde** [comment]
oh, and btw: do I have to do anything analysis-wise?

**Comment by NeptuneHub** [comment]
No, no new analysis needed. Was just a wrong (too generic) select.

**Comment by NeptuneHub** [comment]
I asusme this is solved and I'm going to close. In case of any other feedback feel free to open a new issue. thanks.
